go/templates/index.tmpl

```diff
@@ -234,36 +234,20 @@
     if (products.includes('assets')) {
       $('#assets').show();
     }
-
-    var linkHandlerCommonOptions = {
-      apiVersion: 'v2',
-      clientName: 'Plaid Quickstart',
-      env: '{{ .plaid_environment }}',
-      product: products,
-      key: '{{ .plaid_public_key }}',
-      countryCodes: '{{ .plaid_country_codes }}'.split(','),
-    };
-    var oauthRedirectUri = '{{ .plaid_oauth_redirect_uri }}';
-    if (oauthRedirectUri != '') {
-      linkHandlerCommonOptions.oauthRedirectUri = oauthRedirectUri;
-    }
-    var oauthNonce = '{{ .plaid_oauth_nonce }}';
-    if (oauthNonce != '') {
-      linkHandlerCommonOptions.oauthNonce = oauthNonce;
-    }
     // This functionality is only relevant for the UK Payment Initiation product.
     if (products.includes('payment_initiation')) {
       $('.payment_initiation').show();
-      $.post('/set_payment_token', {}, function(data) {
+      $.post('/create_link_token_with_payment', {}, function(data) {
         console.log(data);
         var paymentToken = data.payment_token;
         $('#payment_token').text(paymentToken);
 
+        localStorage.setItem('link_token', data.link_token);
         // In the case of payment_initiation product, we need to wait for
-        // payment token to be generated before the Link handler can be
-        // initialized.
+        // link token (with associated paymetn token) to be generated before the
+        // Link handler can be initialized.
         handler = Plaid.create({
-          ...linkHandlerCommonOptions,
+          token: data.link_token,
           paymentToken: paymentToken,
```

Comment:
I believe we should be removing `paymentToken` here

Reply:
good catch